hw10-byteCodes/src/main/java/homework/ProxyDemo.java

```diff
@@ -0,0 +1,13 @@
+package homework;
+
+import homework.impl.TestLoggingImpl;
+import homework.proxy.Ioc;
+
+public class ProxyDemo {
+    public static void main(String[] args) {
+        TestLoggingInterface myClass = (TestLoggingInterface) Ioc.createMyClass(new TestLoggingImpl());
```

Comment:
Приведение типа здесь выглядит некрасиво:
```suggestion
        TestLoggingInterface myClass =  Ioc.createMyClass(new TestLoggingImpl());
```

`Ioc.createMyClass()` надо было оставить привязанным к `TestLoggingInterface` - там вы создаете только его:
```
    public static TestLoggingInterface createMyClass(Object myClass) {
        InvocationHandler handler = new DemoInvocationHandler<>(myClass);
        return (TestLoggingInterface)Proxy.newProxyInstance(
                Ioc.class.getClassLoader(), myClass.getClass().getInterfaces(), handler);
    }
```

А с `DemoInvocationHandler` всё правильно. Он получился универсальным и его можно использовать для других класса, интерфейса.

Reply:
Понятно